### PR TITLE
fix: fix second inference error

### PR DESCRIPTION
**Pull request by lucky9-cyou**
Fix second inference error https://github.com/evilsocket/cake/issues/26.

This error was caused by not properly clearing the kv cache on the worker node, so I added a goodbye message from the master node to clear the kv cache on the worker nodes.

And some other update is caused by `cargo fmt`

### CODE REVIEW

File: cake-core/src/cake/mod.rs [hunk 1]
```diff
@@ -123,19 +129,23 @@ pub trait Forwarder: Debug + Send + Sync + Display {
...
         &mut self,
         _x: &Tensor,
         _batch: Vec<(String, usize, usize)>,
-        _ctx: &mut Context
+        _ctx: &mut Context,
     ) -> Result<Tensor> {
         unimplemented!()
     }
 
+    async fn goodbye(&mut self) -> Result<()> {
```
Comment (lucky9-cyou) [inline cake-core/src/cake/mod.rs:145]: This function

**Comment by chosen-ox** [comment]
https://github.com/evilsocket/cake/blob/5a3555cae63d586787391cba83254a7a78d5e387/cake-core/src/models/llama3/attention.rs#L108
Here does not consider the size of kvcache. We can modify it to support history.

**Comment by evilsocket** [comment]
@lucky9-cyou thank you so much! this PR is pretty big, is there any way you can send it without the reformatting and the dependencies updates so I can clearly see the scope of what changed functionally?

**Comment by lucky9-cyou** [comment]
Sure, I will open another PR

**Comment by lucky9-cyou** [comment]
please ref: https://github.com/evilsocket/cake/pull/32

**Comment by evilsocket** [comment]
@lucky9-cyou thank you so much, again! merged

**Pull request closed without merging**